### PR TITLE
Fixed filters. Tested for both JSON and SQL databases.

### PR DESCRIPTION
**Pull request by mishashin**
@shashinadya, please test it locally for both databases. Use positive and negative tests to ensure it works properly in different scenarios.

### CODE REVIEW

**Comment by mishashin** [comment]
Pool request for [44](https://github.com/shashinadya/code-practice/issues/44)

File: database/src/main/java/database/controller/DatabaseServiceRestController.java [hunk 1]
```diff
@@ -122,7 +122,28 @@ void handleAddNewRecord(Context ctx) {
...
+            path = "/api/v1/database/{entityClass}/filter",
+            methods = HttpMethod.GET,
+            pathParams = {
+                    @OpenApiParam(name = "entityClass", description = "Class of the entity", required = true)
+            },
+            queryParams = {
+                    @OpenApiParam(name = "Format: fieldName=value, separated by &",
+                            description = "Filters by entity's fields. Multiple values for one field can be provided " +
+                                    "using the following pattern: filter1=value1&filter1=value2"),
+                    @OpenApiParam(name = "offset", description = "Offset value from the first record")
```
Comment (shashinadya) [inline database/src/main/java/database/controller/DatabaseServiceRestController.java]: Why do we need offset parameter description here, if it's not a filter for the path "/api/v1/database/{entityClass}/filter"? We can use offset as parameter only for this path "/api/v1/database/{entityClass}".

File: database/src/main/java/database/helper/Validator.java [hunk 2]
```diff
@@ -2,40 +2,34 @@
...
-                    .findFirst()
-                    .orElseThrow(() -> new IncorrectPropertyNameException("Incorrect property name: " + propertyName));
-
-            if (!matchingField.getType().isInstance(expectedValue)) {
-                throw new IncorrectValueTypeException("Incorrect value type for filter: " + propertyName
-                        + ". Expected: " + matchingField.getType().getName() + ", but got: " +
-                        expectedValue.getClass().getName());
-            }
+                    .findAny()
+                    .orElseThrow(() -> new IncorrectPropertyNameException(INCORRECT_FILTER_NAME_MESSAGE + ": " + propertyName));
```
Comment (shashinadya) [inline database/src/main/java/database/helper/Validator.java]: Too long line, wrap it to another line please.

File: database/src/main/java/database/controller/DatabaseServiceRestController.java (hunk 1 above)
Comment (mishashin) [reply database/src/main/java/database/controller/DatabaseServiceRestController.java]: Good catch, forgot to remove this.

File: database/src/main/java/database/helper/Validator.java (hunk 2 above)
Comment (mishashin) [reply database/src/main/java/database/helper/Validator.java]: Fixed.

**Review by shashinadya — APPROVED** [review]

**Pull request merged**